apps/frontend/src/hooks/algolia.ts

```diff
@@ -1,26 +1,46 @@
 import { useEffect, useState } from 'react';
-import algoliasearch from 'algoliasearch/lite';
 import { useCurrentUser } from '@asap-hub/react-context';
+import {
+  algoliasearchLite,
+  SearchClientLite,
+  SearchIndex,
+  ResearchOutputSearchIndex,
+} from '@asap-hub/algolia';
 import { User } from '@asap-hub/auth';
 
 import { ALGOLIA_APP_ID, ALGOLIA_INDEX } from '../config';
 
+export type AlgoliaHook = {
+  client: SearchClientLite;
+  index: {
+    researchOutput: ResearchOutputSearchIndex;
+  };
+};
+
 export const useAlgolia = () => {
-  const initAlgolia = (user: User | null) => {
+  const initAlgolia = (user: User | null): AlgoliaHook => {
     if (!user) {
       throw new Error('Algolia unavailable while not logged in');
     }
-    const client = algoliasearch(ALGOLIA_APP_ID, user.algoliaApiKey);
-    const index = client.initIndex(ALGOLIA_INDEX);
+
+    const client = algoliasearchLite(ALGOLIA_APP_ID, user.algoliaApiKey);
+    const researchOutput = new ResearchOutputSearchIndex(
+      client.initIndex(ALGOLIA_INDEX) as SearchIndex,
```

Comment:
Why are we casting here?

Reply:
due to it's taken from algoliasearch/lite and not algoliasearch.
Sth wrong with types on this algoliasearch lib